### PR TITLE
Hide empty reward descriptions

### PR DESCRIPTION
**Pull request by coryroy**
## What
- Hide reward descriptions that are empty as people are now not required to have one

## How
- Added a subscriber that mapped to isEmpty
```java
viewModel.outputs.descriptionTextViewText()
      .compose(bindToLifecycle())
      .compose(observeForUI())
      .map(String::isEmpty)
      .subscribe(ViewUtils.setGone(descriptionTextView));
```

### CODE REVIEW

File: app/src/main/java/com/kickstarter/viewmodels/RewardViewModel.java [hunk 1]
```diff
@@ -378,4 +385,8 @@ private static boolean isSelectable(final @NonNull Project project, final @NonNu
   @Override public @NonNull Observable<Boolean> whiteOverlayIsHidden() {
     return whiteOverlayIsHidden;
   }
+
+  @Override public @NonNull Observable<Boolean> rewardDescriptionIsHidden() {
```
Comment (mbrandonw) [inline app/src/main/java/com/kickstarter/viewmodels/RewardViewModel.java]: can remove this now that we moved to the view model

File: app/src/main/java/com/kickstarter/viewmodels/outputs/RewardViewModelOutputs.java [hunk 2]
```diff
@@ -137,4 +137,9 @@
    * Returns `true` if the white overlay indicating a reward is disabled should be hidden, `false` otherwise.
    */
   Observable<Boolean> whiteOverlayIsHidden();
+
+  /**
+   * Returns `true` if the reward description is empty and should be hidden in the UI
+   */
+  Observable<Boolean> rewardDescriptionIsHidden();
```
Comment (mbrandonw) [inline app/src/main/java/com/kickstarter/viewmodels/outputs/RewardViewModelOutputs.java]: can ya alphabetize this?

File: app/src/test/java/com/kickstarter/factories/RewardFactory.java [hunk 3]
```diff
@@ -81,4 +81,11 @@ private RewardFactory() {}
       .minimum(1.0f)
       .build();
   }
+
+  public static @NonNull Reward noDescription() {
+    return reward().toBuilder()
+      .description("")
+      .build();
+
```
Comment (mbrandonw) [inline app/src/test/java/com/kickstarter/factories/RewardFactory.java]: extra lil  newline here

File: app/src/main/java/com/kickstarter/viewmodels/outputs/RewardViewModelOutputs.java (hunk 2 above)
Comment (coryroy) [inline app/src/main/java/com/kickstarter/viewmodels/outputs/RewardViewModelOutputs.java]: I will fix these two items and get in touch via slack

File: app/src/main/java/com/kickstarter/viewmodels/RewardViewModel.java (hunk 1 above)
Comment (coryroy) [inline app/src/main/java/com/kickstarter/viewmodels/RewardViewModel.java]: um, this is the view model.

Comment (mbrandonw) [inline app/src/main/java/com/kickstarter/viewmodels/RewardViewModel.java]: lol nvm! was confused

**Pull request merged**